apps/meteor/app/file-upload/server/lib/FileUpload.ts

```diff
@@ -308,9 +308,18 @@ export const FileUpload = {
 		const store = FileUpload.getStore('Uploads');
 		const image = await store._store.getReadStream(file._id, file);
 
-		const transformer = sharp().resize({ width, height, fit: 'inside' });
-
-		const result = transformer.toBuffer({ resolveWithObject: true }).then(({ data, info: { width, height } }) => ({ data, width, height }));
+		let transformer = sharp().resize({ width, height, fit: 'inside' });
+
+		if (file.type === 'image/svg+xml') {
+			transformer = transformer.png();
+			file.name = (file.name || '').replace(/\.svg$/i, '.png');
```

Comment:
IMO we shouldn't do this. Just add the extension if you want for clarity. as the extensions mean nothing, we shouldn't touch the original file name either.